### ISSUE TITLE
firefox nightly: weird grey top bar & reload button randomly added to uBO's extension UI

### ISSUE THREAD

**Issue by ComfortableMilk4454** [uBlockOrigin/uBlock-issues#3284]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

  firefox nightly: weird grey top bar that disappears after a few seconds and when that happens reload button gets randomly added to uBO's extension UI (see [screen recording](https://youtu.be/LfvurCkgl-s)): ***THIS ONLY HAPPENS UPON LAUNCHING NIGHTLY & CLICKING ON UBO'S EXTENSION ICON, NOT WHEN NAVIGATING TO A NEW TAB WITH NIGHTLY ALREADY RUNNING & CLICKING ON UBO'S EXTENSION ICON***. fyi if you saw i have adguard running on my mac dw i made sure to turn it off for firefox, so this issue is not caused by "stacking" adblockers

### A specific URL where the issue occurs.

```yaml
new nightly tab
```

### Steps to Reproduce

1. click on uBO's icon

### Expected behavior

no bar at the top & no refresh button appearing

### Actual behavior

bar at the top & refresh button appears

### uBO version

1.58.0

### Browser name and version

Firefox Nightly 129.0a1 (2024-06-28) (64-bit) & all versions of nightly over the past month or so

### Operating System and version

macOS 14.5 (23F79)

**Comment by ComfortableMilk4454** [comment]
i see. so this is a nightly issue then, because this doesn't happen on firefox release, right? thanks for explaining.

**Comment by gorhill** [comment]
Not necessarily a Nightly issue. There is a lot of thing which can prevent uBO's background process from answering in a timely manner to requests from its auxiliary page. In Firefox all extensions execute in the same process, so even another extension hogging the webextensions process could affect responsiveness in other extensions. It could be a delay originating from Firefox itself, for example <https://bugzilla.mozilla.org/show_bug.cgi?id=1899015>.

Profiling at browser launch would maybe help identify the sequence of events. In any case, there is no malfunction, it's just that there is a delay between the popup requesting information from the background process, and the background process answering that delay.

I could add a non-breakable whitespace character as placeholder to ensure the vertical size of the grey bar does not change.

**Comment by garry-ut99** [comment]
Not only a Firefox Nightly issue, as the same occurs also on Firefox Release on a fresh browser profile.
Currently doesn't occur in Chromium and Edge, so it's related to Firefox.
However it was happening also in Chromium and Edge in the past, but not anymore, as still can be reproduced in old browsers like Chromium 109 or Edge 109.

> this only happens upon launching nightly & clicking on ubo's extension icon

Or by turning off & on uBlock in browser extension pane (Firefox).
In old Chromium 109 or Edge 109, I still need to use "restarting browser" method to reproduce.
Method might depend on how fast is CPU, as different users have different CPUs.

<hr>

I see also big blue button being white for a while in the both issues, might be related or similiar or duplicate of:
- https://github.com/uBlockOrigin/uBlock-issues/issues/757

**Comment by ComfortableMilk4454** [comment]
@gorhill was this fixed by chance with [this commit](https://github.com/gorhill/uBlock/commit/8c5918dec78f7217103eace03ec6f2c376224b90)? i cant read code thats why im asking lol

**Comment by garry-ut99** [comment]
Yes, the commit is related, exactly fixed this: "weird grey top bar" :

>  I could add a non-breakable whitespace character as placeholder to ensure the vertical size of the grey bar does not change.

(But not "Reload button".)

**Comment by ComfortableMilk4454** [comment]
thanks for the response @garry-ut99. is this only fixed for firefox release? cuz im on nightly 30.0a1 (2024-07-09) (64-bit) & it still is happening. uBO 1.58.0.

https://github.com/uBlockOrigin/uBlock-issues/assets/43555540/2b8191db-61dc-4fec-9b90-3f05b37e9c0d

**Comment by stephenhawk8054** [comment]
@ComfortableMilk4454 The fix is in uBO dev build (`1.58.1rc0`), not the current stable version (`1.58.0`)

**Comment by ComfortableMilk4454** [comment]
@stephenhawk8054 that would explain it lol. ill wait for the stable extension to be updated. thanks for letting me know!

**Comment by garry-ut99** [comment]
> ill wait for the stable extension to be updated. 

You can, but you don't need to, you can install dev build to test: https://github.com/gorhill/uBlock/releases/tag/1.58.1rc0, just click `xpi` file.

**Comment by ComfortableMilk4454** [comment]
thanks but id rather wait cuz i prefer stable builds to dev builds for the stability. ill let you know once 1.58.1 comes out on ff addons if its fixed.

**Comment by garry-ut99** [comment]
As you wish, I only mentioned it in case you didn't know, also you can install "uBO Dev" in a separate browser profile for testing purposes, I didn't tell you to overwrite your "uBO Release".

**Comment by ComfortableMilk4454** [comment]
no worries, i figured thats why u mentioned it cuz probs a lot of ppl dont know they can install the dev build, but i happened to know that already :). sure ill test on my firefox release rn with the dev build & let you know

**Comment by ComfortableMilk4454** [comment]
@garry-ut99 oh damn it doesn't work with 1.58.1rc0, i used this link to get it https://github.com/gorhill/uBlock/releases/download/1.58.1rc0/uBlock0_1.58.1rc0.firefox.signed.xpi. ff release 128.0, any suggestions?

https://github.com/uBlockOrigin/uBlock-issues/assets/43555540/2c5e2cb6-88ea-4820-8ca2-9c04d228aa2d

**Comment by garry-ut99** [comment]
Yes, seems still unfixed, even with uBO 1.58.1rc0.
(or fixed in some cases, but not in all cases).

**Comment by ComfortableMilk4454** [comment]
@gorhill ^

**Comment by gorhill** [comment]
Marking as wontfix then, not worth spending time on what is merely a visual artefact occurring once when the extension is first installed.

**Comment by ComfortableMilk4454** [comment]
@gorhill it occurs everytime you launch ff but thats fair, thanks for trying anyways :)

**Comment by gorhill** [comment]
Alright, let me rephrase then:

not worth spending time on what is merely a visual artefact when clicking too early on the toolbat icon when the browser is launched.

**Comment by ComfortableMilk4454** [comment]
Oh ok lol, got it. Thanks for uBO btw, best extension I’ve ever had :)

**Comment by krystian3w** [comment]
Possible use 0 width space `&#8203;` / `&#x200b;`.

Then the only problem is when the domain has more than 60 characters in the database or more than 3 in the TLD or is extension address as homepage/start page (these combinations may need 2 or more lines and are too rare to address).
